### ISSUE TITLE
Objects from other mods don't always render properly.

### ISSUE THREAD

**Issue by Keridos** [CptSpaceToaster/CptsModdingLight#26]
The Thaumcraft Aura nodes render almost black with this mod installed.

**Comment by CptSpaceToaster** [comment]
There are a large number of rendering issues.

I can't fix all of them, but I am aware of quite a couple.  I'm looking into any possibility of fixing these issues from my end... but it's often consequences of the changes I made to the rendering engine.  The lighting engine was backwards compatible, but the rendering engine is "mostly" backwards compatible.  Leaving it fragile, and easy to break.

Consult the [list](https://docs.google.com/spreadsheets/d/1mfAradGBs1U2LOVHIW6_2FNKnc4pKB00qr9VAYtrKZc/edit?usp=sharing).  Please add an entry, or edit a description if you don't see your issue.

Pull-Requests are welcome.

**Comment by Keridos** [comment]
I had a ton of other issues as well, but I am running a large modpack. Basically nothing worked for me, Lights where no lights were placed, but I had torches giving no light at all and my basement was pretty much randomly colored, no idea what went on there.

/edit: I have fastcraft in there, pretty sure that messed that up though.

**Comment by CptSpaceToaster** [comment]
Yeah, one of the larger issues is that the lighting engine expects light to have "color values" but by default, vanilla sets them to zero.

This means that the lighting engine is still backwards compatible, but the light will be colored black.  Some of the larger offenders are Twilight Forests' little glow bugs, and too many others to mention.

I'll have to create a basic interface that records the offenders, and creates a hook that asks my mod for lightvalues.  Then, I can correct the color values (or change them if I want :D) which would fix some of the "missing lights"

To clarify, the vanilla lightvalue should still be spreading around, it just won't look like it's rendering.  Check your F3 menu to be sure.

**Comment by Keridos** [comment]
Yeah, i figured that it should not alter the vanilla light values since a lot more stuff depends on those. Too bad we use fastcraft on our private server, though, cannot put CLC on there as it is now.
Btw, I asked player in the fastcraft forums if he could enable compatability to this mod somehow.

**Comment by CptSpaceToaster** [comment]
I sent him a message.

**Comment by Keridos** [comment]
http://forum.industrial-craft.net/index.php?page=Thread&postID=178589#post178589

**Comment by CptSpaceToaster** [comment]
I read through some of the history, and he claimed that 1.16 made it compatible... Are you running that version but still experiencing issues?

Because I have another user who confirmed that 1.16 worked.

**Comment by Keridos** [comment]
No, our server is still at 1.11 or 1.12

**Comment by CptSpaceToaster** [comment]
Read this post then: http://forum.industrial-craft.net/index.php?page=Thread&postID=178558#post178558

They already added support. (Thank you player!)

**Comment by CptSpaceToaster** [comment]
I may recreate this issue to have more relevant information, discussing issues with other mod renderers, and the pitfalls that can occur.

**Comment by CptSpaceToaster** [comment]
Some documentation that explains on a technical level, why things turn blue:
https://gist.github.com/CptSpaceToaster/8b6dbd107e5fe44ab7b7
